presto-tests/src/test/java/io/prestosql/execution/TestFinalQueryInfo.java

```diff
@@ -63,25 +63,26 @@ private static QueryId startQuery(String sql, DistributedQueryRunner queryRunner
     {
         OkHttpClient httpClient = new OkHttpClient();
         try {
-            ClientSession clientSession = new ClientSession(
-                    queryRunner.getCoordinator().getBaseUrl(),
-                    "user",
-                    "source",
-                    Optional.empty(),
-                    ImmutableSet.of(),
-                    null,
-                    null,
-                    null,
-                    null,
-                    ZoneId.of("America/Los_Angeles"),
-                    Locale.ENGLISH,
-                    ImmutableMap.of(),
-                    ImmutableMap.of(),
-                    ImmutableMap.of(),
-                    ImmutableMap.of(),
-                    ImmutableMap.of(),
-                    null,
-                    new Duration(2, MINUTES));
+            ClientSession clientSession = ClientSession.builder()
+                    .withServer(queryRunner.getCoordinator().getBaseUrl())
+                    .withUser("user")
+                    .withSource("source")
+                    .withTraceToken(Optional.empty())
+                    .withClientTags(ImmutableSet.of())
+                    .withClientInfo(null)
+                    .withCatalog(null)
+                    .withSchema(null)
+                    .withPath(null)
+                    .withTimeZone(ZoneId.of("America/Los_Angeles"))
+                    .withLocale(Locale.ENGLISH)
+                    .withResourceEstimates(ImmutableMap.of())
+                    .withProperties(ImmutableMap.of())
+                    .withPreparedStatements(ImmutableMap.of())
+                    .withRoles(ImmutableMap.of())
+                    .withCredentials(ImmutableMap.of())
+                    .withoutTransactionId()
```

Comment:
is this required?

Reply:
yes, there is a requireNonNull check in ClientSession